### ISSUE TITLE
Day1

### ISSUE THREAD

**Issue by xiaohongniua** [Avik-Jain/100-Days-Of-ML-Code#13]
is it  better to do feature scaler before dataset split ?

**Comment by wengJJ** [comment]
那样就失去划分测试集训练集的意义了 测试集训练集是不能有联系的

**Comment by xiaohongniua** [comment]
thanks,I am a new learner too,so there may be  some wrong, lets study together.

I found some examples,it seemed that we use scaler.fit_transform(X_train) first,then scaler.transform(X_test),the meaning is to fit the scaler by X_train first and then use this scalar to transform X_test (though the code here might be wrong,here is scaler.fit_transform(X_test)),,,  so I am wondering if we can do  feature scaler before dataset split.

what you said might not be totally right. It seems that we do preditction under the hypotheses that the  distribution of train_data and test_data are similiar, actually,doing feature scaler before or after dataset split are equally.

**Comment by wengJJ** [comment]
I'm a new learner,too.
For example,**LinearNormalization:f(x) = (x - min) / (max - min)**.There are maximum and minimum values involved.if we use do feature scaler before dataset split.Perhaps the maximum or minimum value exists in the test set.
But in fact,It's like we're predicting the stock price tomorrow,We don't know the stock price tomorrow,so If we want to train the model, we can only use the previous data, and feature scaling can only use the previous data.

**Comment by wengJJ** [comment]
your example is right,just you said,we can use the X_train-scalar to transform X_test.But we also need to split dataset first.

**Comment by sonumja** [comment]
We can scale data before or after splitting.
Feature Scaling before splitting:
from sklearn.preprocessing import StandardScaler
sc_X = StandardScaler()
sc_y = StandardScaler()
X = sc_X.fit_transform(X)
y = sc_y.fit_transform(y)

Feature scaling after splitting:
from sklearn.preprocessing import StandardScaler
sc_X = StandardScaler()
X_train = sc_X.fit_transform(X_train)
X_test = sc_X.transform(X_test)
sc_y = StandardScaler()
y_train = sc_y.fit_transform(y_train)

**Comment by dieee** [comment]
cross_validation.py:41: DeprecationWarning: This module was deprecated in version 0.18 in favor of the model_selection module into which all the refactored classes and functions are moved. Also note that the interface of the new CV iterators are different from that of this module. This module will be removed in 0.20.
  "This module will be removed in 0.20.", DeprecationWarning)
